src/status_im2/contexts/chat/bottom_sheet_composer/effects.cljs

```diff
@@ -0,0 +1,91 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.effects
+  (:require
+    [react-native.async-storage :as async-storage]
+    [react-native.core :as rn]
+    [react-native.reanimated :as reanimated]
+    [status-im2.contexts.chat.bottom-sheet-composer.constants :as c]
+    [status-im2.contexts.chat.bottom-sheet-composer.keyboard :as kb]
+    [utils.re-frame :as rf]))
+
+(defn reenter-screen-effect
+  [{:keys [text-value saved-cursor-position maximized?]}
+   {:keys [content-height]}
+   {:keys [input-content-height input-text input-maximized?]}]
+  (when (and (empty? @text-value) (not= input-text nil))
+    (reset! text-value input-text)
+    (reset! content-height input-content-height)
+    (reset! saved-cursor-position (count input-text)))
+  (when input-maximized?
+    (reset! maximized? true)))
+
+(defn maximized-effect
+  [{:keys [maximized?]}
+   {:keys [height saved-height last-height]}
+   {:keys [max-height]}
+   {:keys [input-content-height]}]
+  (when (or @maximized? (>= input-content-height max-height))
+    (reanimated/animate height max-height)
+    (reanimated/set-shared-value saved-height max-height)
+    (reanimated/set-shared-value last-height max-height)))
+
+(defn refocus-effect
+  [{:keys [input-ref]}
+   {:keys [input-refocus?]}]
+  (when input-refocus?
+    (.focus ^js @input-ref)
+    (rf/dispatch [:chat.ui/set-input-refocus false])))
+
+(defn layout-effect
+  [{:keys [lock-layout?]}]
+  (when-not @lock-layout?
+    (js/setTimeout #(reset! lock-layout? true) 500)))
+
+(defn kb-default-height-effect
+  [{:keys [kb-default-height]}]
+  (when-not @kb-default-height
+    (async-storage/get-item :kb-default-height
+                            #(reset! kb-default-height (when (not= nil %) (js/parseInt %))))))
```

Comment:
This can be
`(not= nil` -> `(some? nil`

Reply:
I'd go further:
If we're not going to receive `false` (I guess no since we're using `js/parseInt`), we could do:

```clojure
(when %
  (js/parseInt %)
```

or

```clojure
(some-> %  (js/parseInt)
```